nearcore/src/lib.rs

```diff
@@ -116,39 +114,31 @@ fn create_db_checkpoint(path: &Path, near_config: &NearConfig) -> Result<PathBuf
 }
 
 /// Function checks current version of the database and applies migrations to the database.
-pub fn apply_store_migrations(path: &Path, near_config: &NearConfig) {
+fn apply_store_migrations(path: &Path, near_config: &NearConfig) -> anyhow::Result<()> {
     let db_version = get_store_version(&path);
-    if db_version > near_primitives::version::DB_VERSION {
-        error!(target: "near", "DB version {} is created by a newer version of neard, please update neard or delete data", db_version);
-        std::process::exit(1);
-    }
-
     if db_version == near_primitives::version::DB_VERSION {
-        return;
+        return Ok(());
     }
+    anyhow::ensure!(
+        db_version < near_primitives::version::DB_VERSION,
+        "DB version {db_version} is created by a newer version of \
+                     neard, please update neard or delete data"
+    );
 
     // Before starting a DB migration, create a consistent snapshot of the database. If a migration
     // fails, it can be used to quickly restore the database to its original state.
     let checkpoint_path = if near_config.config.use_db_migration_snapshot {
-        match create_db_checkpoint(path, near_config) {
-            Ok(checkpoint_path) => {
-                info!(target: "near", "Created a DB checkpoint before a DB migration: '{}'. Please recover from this checkpoint if the migration gets interrupted.", checkpoint_path.display());
-                Some(checkpoint_path)
-            }
-            Err(err) => {
-                panic!(
-                    "Failed to create a database migration snapshot:\n\
-                     {}\n\
-                     Please consider fixing this issue and retrying.\n\
-                     You can change the location of database migration snapshots by adjusting `config.json`:\n\
-                     \t\"db_migration_snapshot_path\": \"/absolute/path/to/existing/dir\",\n\
-                     Alternatively, you can disable database migration snapshots in `config.json`:\n\
-                     \t\"use_db_migration_snapshot\": false,\n\
-                     ",
-                    err
-                );
-            }
-        }
+        let checkpoint_path = create_db_checkpoint(path, near_config)
+            .map_err(|err| anyhow::anyhow!(
```

Comment:
I think you want to use `.context` rather than `map_err` here. 

https://docs.rs/anyhow/latest/anyhow/trait.Context.html#example